generators/generator-base.js

```diff
@@ -1517,6 +1517,40 @@ module.exports = class extends PrivateBase {
         this.copyTemplate(source, dest, 'stripHtml', generator, opt, template);
     }
 
+    /**
+     * Copy Ts templates after stripping translation keys when translation is disabled.
+     *
+     * @param {string} source - path of the source file to copy from
+     * @param {string} dest - path of the destination file to copy to
+     * @param {object} generator - context that can be used as the generator instance or data to process template
+     * @param {object} opt - options that can be passed to template method
+     * @param {boolean} template - flag to use template method instead of copy
+     */
+    processTs(source, dest, generator, opt, template) {
+        this.copyTemplate(source, dest, 'stripJs', generator, opt, template);
+        if (this.configOptions.codeFormatter) {
+            this.log(`prettier: ${chalk.green(dest)}`);
+            const output = jhipsterUtils.prettierFormat(generator, dest, 'prettierrc.json');
+            generator.fs.write(dest, output);
+        }
+    }
+
+
+    /**
+     * Copy config files templates directly on the disk to use it to configure while generating
+     *
+     * @param {string} source - path of the source file to copy from
+     * @param {string} dest - path of the destination file to copy to
+     * @param {object} generator - context that can be used as the generator instance or data to process template
+     * @param {object} opt - options that can be passed to template method
+     * @param {boolean} template - flag to use template method instead of copy
+     */
+    processConfig(source, dest, generator, opt, template) {
```

Comment:
why is this method needed? I dont understand the significance of using this for the prettier config file

Reply:
To use prettier while generating file i need to give the function the path to the config file
But i did not found the path of this file before the copy so i force the generator to copy it before all of the other files i want to format